File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/javav2/example_code/emr/src/main/java/aws/example/emr/TerminateJobFlow.java
/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/
package aws.example.emr;

import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.emr.EmrClient;
import software.amazon.awssdk.services.emr.model.*;
import software.amazon.awssdk.services.emr.model.TerminateJobFlowsRequest;

/*
 *   Ensure that you have setup your development environment, including your credentials.
 *   For information, see this documentation topic:
 *
 *   https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
 *
 */
public class TerminateJobFlow {

    public static void main(String[] args){

        final String USAGE = "\n" +
                "Usage: " +
                "   <id>\n\n" +
                "Where:\n" +
                "   id - an id of a job flow to shut down.\n\n" ;

        if (args.length != 1) {
              System.out.println(USAGE);
              System.exit(1);
          }


        String id = args[0] ;
        Region region = Region.US_WEST_2;
        EmrClient emrClient = EmrClient.builder()
                .region(region)
                .build();

        terminateFlow(emrClient, id);
         emrClient.close();
    }

    public static void terminateFlow( EmrClient emrClient, String id) {

        try{

            TerminateJobFlowsRequest jobFlowsRequest = TerminateJobFlowsRequest.builder()
                    .jobFlowIds(id)
                    .build();

            emrClient.terminateJobFlows(jobFlowsRequest);
            System.out.println("You have successfully terminated "+id);

        } catch(EmrException e){
            System.err.println(e.getMessage());
            System.exit(1);
        }
    }
}

File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/kotlin/services/emr/src/main/kotlin/com/kotlin/emr/TerminateJobFlow.kt
/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.kotlin.emr

import aws.sdk.kotlin.services.emr.EmrClient
import aws.sdk.kotlin.services.emr.model.TerminateJobFlowsRequest
import kotlin.system.exitProcess


/**
To run this Kotlin code example, ensure that you have set up your development environment,
including your credentials.

For information, see this documentation topic:
https://docs.aws.amazon.com/sdk-for-kotlin/latest/developer-guide/setup.html
 */

suspend fun main(args:Array<String>) {

    val usage = """
          Usage:    
            <id>
    
          Where:
            id - an id of a job flow to shut down. 
        """

     if (args.size != 1) {
          System.out.println(usage)
          exitProcess(1)
     }

    val id = args[0]
    terminateFlow(id)
}

suspend fun terminateFlow(id: String) {

    val request = TerminateJobFlowsRequest{
        jobFlowIds = listOf(id)
    }

    EmrClient { region = "us-west-2" }.use { emrClient ->
        emrClient.terminateJobFlows(request)
        println("You have successfully terminated $id")
    }
}

